examples/linear_model/plot_lasso_and_elasticnet.py

```diff
@@ -1,97 +1,245 @@
 """
-========================================
-Lasso and Elastic Net for Sparse Signals
-========================================
+==================================
+L1-based models for Sparse Signals
+==================================
 
-Estimates Lasso and Elastic-Net regression models on a manually generated
-sparse signal corrupted with an additive noise. Estimated coefficients are
-compared with the ground-truth.
+The present example compares three l1-based regression models on a synthetic
+signal obtained from sparse and correlated features that are further corrupted
+with additive gaussian noise:
+
+ - a :ref:`lasso`;
+ - an :ref:`automatic_relevance_determination`;
+ - an :ref:`elastic_net`.
+
+It is known that the Lasso estimates turn to be close to the model selection
+estimates when the data dimensions grow, given that the irrelevant variables are
+not too correlated with the relevant ones. In the presence of correlated
+features, Lasso itself cannot select the correct sparsity pattern [1]_.
 
+Here we compare the performance of the three models in terms of the :math:`R^2`
+score, the fitting time and the sparsity of the estimated coefficients when
+compared with the ground-truth.
 """
 
+# Author: Arturo Amor <david-arturo.amor-quiroz@inria.fr>
+
 # %%
-# Data Generation
-# ---------------------------------------------------
+# Generate synthetic dataset
+# --------------------------
+#
+# We generate a dataset where the number of samples is lower than the total
+# number of features. This leads to a singular matrix that cannot be dealt with
+# an :ref:`ordinary_least_squares` as it requires some regularization.
```

Comment:
I am not sure about the statement here.

It leads to a singular matrix and it is impossible to compute the inverse when intending to make an OLS. Regularization is just a technique that will allow to inverse the matrix because it makes non-singular.
So I am not sure that "as it requires" is really what we want here.